server/controllers/user.js

```diff
@@ -19,14 +14,27 @@ function allUsers(req, res) {
 function updateUser(req, res) {
     let db = req.app.get('db');
     let { id, updates } = req.body;
-    // TO-DO check permission here to see if user is updating themself or someone else (or if it's an admin)
-    // i.e. get the logged in user info by session cookie and see if they are an admin, or if they are editing themself
+    let loggedInUser = req.session.user;
+
+    // keeps someone that's not an admin from doing things they shouldn't
+    if (loggedInUser.access_level < 10) {
+        if (loggedInUser.id !== id)
+            return sendFailure(res, 'Cannot edit another user');
+        delete updates.access_level;
+        delete updates.user_type;
+    }
+
     return db.users.update({ id }, updates)
+        .then(user => sendSuccess(res, user[0]))
+        .catch(e => sendError(res, e))
 }
 
-// Used to add/revome someone's cashier priviledges
-function updatePermissions(req, res) {
-
+function getUserById(req, res) {
+    let db = req.app.get('db');
+    let { id } = req.body;
+    return db.users.find({ id })
+        .then(user => sendSuccess(res, user[0]))
+        .catch(e => sendError(res, e))
 }
 
```

Comment:
got rid of updatePermissions, since updateUser can do the same thing. updateUser checks that it's an admin before allowing permissions to be updated, so we should be ok to use updateUser to add/remove cashier priviledges